### PR TITLE
Check that session hasn't changed after callout

### PR DESCRIPTION
**Pull request by amstone326**
Clear the current session and return to home screen if the session's needed datum has changed (including being null) when we return from a session-related callout. Provides better handling of this crash: https://commcare-acralyzer.cloudant.com/acralyzer/_design/acralyzer/index.html#/reports-browser/commcare-odk/bug/b08b624100441f02ede2ab35fb0e1bdc

 Also add a popup dialog telling the user what has happened when the session is cleared.

### CODE REVIEW

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -864,7 +883,8 @@ private void handleGetCommand(AndroidSessionWrapper asw) {
         } else {
             i = new Intent(getApplicationContext(), MenuList.class);
         }
-        i.putExtra(SessionFrame.STATE_COMMAND_ID, asw.getSession().getCommand());
+        i.putExtra(SessionFrame.STATE_COMMAND_ID, command);
+        addPendingDatumIdExtra(i, asw.getSession());
```
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: causes and NPE if you select a module that doesn't use cases

**Comment by amstone326** [comment]
@phillipm Fixed, thanks for catching. @ctsims Should probably look at this before merging though.

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 2]
```diff
@@ -565,6 +579,19 @@ else if(resultCode == RESULT_OK){
...
+     * @return If the nature of the data that the session is waiting for has not changed since the
+     * callout that we are returning from was made
+     */
+    private boolean sessionStateUnchangedSinceCallout(CommCareSession session, Intent intent) {
+        boolean dataIdsEqual = session.getNeededData().equals(
+                intent.getStringExtra(KEY_PENDING_SESSION_DATA));
+        String intentDatum = intent.getStringExtra(KEY_PENDING_SESSION_DATUM_ID);
+        boolean datumIdsEqual = intentDatum  == null ? true :
+                intentDatum.equals(session.getNeededDatum().getDataId());
+        return dataIdsEqual && datumIdsEqual;
```
Comment (ctsims) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: These seem very close, it took me two reads to realize they weren't the same value

**Comment by ctsims** [comment]
@amstone326 one minor change then good to go

**Comment by amstone326** [comment]
@ctsims fixed

**Pull request merged**